### PR TITLE
feat: Allow specifying path for button

### LINKED ISSUE #1329: Allow specifying path for button
### Discussed in https://github.com/h2oai/wave/discussions/1318

Add a `path` prop to button which opens an external URL in a new tab after clicking.

### PR DESCRIPTION
**Pull request by marek-mihok**
Adds optional `path` prop to ui.button component:
```ts
export interface Button {
  /** An identifying name for this component. If the name is prefixed with a '#', the button sets the location hash to the name when clicked. */
  name: Id
  /** The text displayed on the button. */
  label?: S
  /** The caption displayed below the label. */
  caption?: S
  /** A value for this button. If a value is set, it is used for the button's submitted instead of a boolean True. */
  value?: S
  /** True if the button should be rendered as the primary button in the set. */
  primary?: B
  /** True if the button should be disabled. */
  disabled?: B
  /** True if the button should be rendered as link text and not a standard button. */
  link?: B
  /** An optional icon to display next to the button label (not applicable for links). */
  icon?: S
  /** The width of the button, e.g. '100px'. */
  width?: S
  /** True if the component should be visible. Defaults to True. */
  visible?: B
  /** An optional tooltip message displayed when a user clicks the help icon to the right of the component. */
  tooltip?: S
  /** The path or URL to link to. */
  path?: S
}
```

Closes #1329

### CODE REVIEW

File: ui/src/button.test.tsx [hunk 1]
```diff
@@ -82,6 +81,17 @@ describe('Button.tsx', () => {
 
       expect(queryByTestId(name)).not.toHaveClass('ms-Link')
     })
+
+    it('Does redirect if path is specified', () => {
+      const
+        windowOpenMock = jest.fn(),
+        btnPathProps: Buttons = { items: [{ button: { name, label: name, path: 'https://h2o.ai/' } }] },
+        { getByTestId } = render(<XButtons model={btnPathProps} />)
```
Comment (mturoci) [inline ui/src/button.test.tsx:89]:
```suggestion
        { getByTestId } = render(<XButtons model={{ items: [{ button: { name, label: name, path: 'https://h2o.ai/' } }] }} />)
```

**Review by mturoci — DISMISSED** [review]
Thanks @marek-mihok. LGTM except 1 small comment.

cc @lo5 for API review - another option is to not open all paths in a new tab by default, but specify a `target`  attr instead.

File: ui/src/button.tsx [hunk 2]
```diff
@@ -62,6 +62,8 @@ export interface Button {
   visible?: B
   /** An optional tooltip message displayed when a user clicks the help icon to the right of the component. */
   tooltip?: S
+  /** The path or URL to link to. */
```
Comment (lo5) [inline ui/src/button.tsx:65]:
```suggestion
  /** The path or URL to link to. If specified, the `name` is ignored. The URL is opened in a new browser window or tab. */
```

**Review by lo5 — APPROVED** [review]
Looks good! Added a suggestion to improve API docstring.

**Pull request merged**